docs/contracts/constant-state-variables.rst

```diff
@@ -18,18 +56,17 @@ The reason behind allowing side-effects on the memory allocator is that it
 should be possible to construct complex objects like e.g. lookup-tables.
 This feature is not yet fully usable.
 
-The compiler does not reserve a storage slot for these variables, and every occurrence is
-replaced by the respective constant expression (which might be computed to a single value by the optimizer).
-
-Not all types for constants are implemented at this time. The only supported types are
-value types and strings.
+Immutable
+=========
 
-::
+Variables declared as ``immutable`` are a bit less restricted than those
+declared as ``constant``: Immutable variables can be assigned an arbitrary
+value in the constructor of the contract or at the point of their declaration.
+They cannot be read during construction time and can only be assigned once.
 
-    pragma solidity >=0.4.0 <0.7.0;
-
-    contract C {
-        uint constant X = 32**22 + 8;
-        string constant TEXT = "abc";
-        bytes32 constant MY_HASH = keccak256("abc");
-    }
+The contract creation code generated by the compiler will modify the
+contract's runtime code before it is returned by inserting all references
```

Comment:
```suggestion
contract's runtime code before it is returned by replacing all references
```

Sorry, this might have been wrong in my suggestion earlier. Alternatively the other way round: "by inserting the values assigned to the immutables into all references to them".